src/backend/mpi/swdsm.h

```diff
@@ -71,38 +76,165 @@ typedef struct myControlData //global cache control data / directory
 } control_data;
 
 /** @brief Struct containing statistics */
-typedef struct argo_statisticsStruct
+typedef struct argo_statistics_struct
 {
-		/** @brief Time spend locking */
-		double locktime;
 		/** @brief Time spent self invalidating */
 		double selfinvtime;
 		/** @brief Time spent loading pages */
-		double loadtime;
+		double load_time;
+		/** @brief Mutex to update load_time */
+		std::mutex load_time_mutex;
 		/** @brief Time spent storing pages */
-		double storetime;
-		/** @brief Time spent writing back from the writebuffer */
-		double writebacktime;
-		/** @brief Time spent flushing the writebuffer */
-		double flushtime;
+		double store_time;
+		/** @brief Mutex to update store_time */
+		std::mutex store_time_mutex;
+		/** @brief Time spent locking the sync lock */
+		double sync_lock_time;
+		/** @brief Mutex to update sync_lock_time */
+		std::mutex sync_lock_time_mutex;
 		/** @brief Time spent in global barrier */
 		double barriertime;
+		/** @brief Time spent initializing ArgoDSM */
+		double inittime;
+		/** @brief Time between init and finalize */
+		double exectime;
 		/** @brief Number of stores */
-		std::size_t stores;
+		std::atomic<std::size_t> write_misses;
 		/** @brief Number of loads */
-		std::size_t loads;
+		std::atomic<std::size_t> read_misses;
 		/** @brief Number of barriers executed */
 		std::size_t barriers;
-		/** @brief Number of writebacks from (full) writebuffer */
-		std::size_t writebacks;
 		/** @brief Number of locks */
 		int locks;
 		/** @brief Time spent performing selective acquire */
-		double ssitime;
+		double ssi_time;
+		/** @brief Mutex to update ssi_time */
+		std::mutex ssi_time_mutex;
 		/** @brief Time spent performing selective release */
-		double ssdtime;
+		double ssd_time;
+		/** @brief Mutex to update ssd_time */
+		std::mutex ssd_time_mutex;
 } argo_statistics;
 
+/**
+ * @brief Simple lock struct
+ * @todo  This should be done in a separate cache class
+ */
+class alignas(64) cache_lock {
+	private:
+		/** 
+		 * @brief Mutex protecting one cache block 
+		 * @todo This should be used through std::scoped_lock
```

Comment:
We _are_ requiring `c++17`, so this comment does not make sense...
So, perhaps it _should_ be done through `std::scoped_lock` !

Reply:
Now using std::scoped_lock where applicable.